src/sunadaptcontroller/soderlind/sunadaptcontroller_soderlind.c

```diff
@@ -112,17 +111,19 @@ SUNAdaptController SUNAdaptController_Soderlind(SUNContext sunctx)
  * Function to set Soderlind parameters
  */
 
-int SUNAdaptController_SetParams_Soderlind(SUNAdaptController C, sunrealtype k1,
-                                           sunrealtype k2, sunrealtype k3,
-                                           sunrealtype k4, sunrealtype k5)
+SUNErrCode SUNAdaptController_SetParams_Soderlind(SUNAdaptController C,
+                                                  sunrealtype k1,
+                                                  sunrealtype k2, sunrealtype k3,
+                                                  sunrealtype k4, sunrealtype k5)
 {
-  if (C == NULL) { return SUNADAPTCONTROLLER_ILL_INPUT; }
+  SUNFunctionBegin(C->sunctx);
+  SUNAssert(C, SUN_ERR_ARG_CORRUPT);
```

Comment:
As in `sunadaptcontroller_imexgus.c`, I see no need for this second line (here, or throughout this file).